src/org/jetbrains/plugins/scala/codeInspection/monads/StatefulMonads.scala

```diff
@@ -0,0 +1,27 @@
+package org.jetbrains.plugins.scala.codeInspection.monads
+
+import com.intellij.psi.PsiClass
+import org.jetbrains.plugins.scala.lang.psi.types.{ScDesignatorType, ScType}
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+ * @author Sergey Tolmachev (tolsi.ru@gmail.com)
+ * @since 29.09.15
+ */
+object StatefulMonads {
+  private[monads] final val RootTypeCanonicalNamePrefix = "_root_."
+  private[monads] final lazy val StatefulMonadsTypes: Set[Class[_]] = Set(classOf[Future[_]], classOf[Try[_]])
+  private[monads] final lazy val StatefulMonadsTypesNames = StatefulMonadsTypes.map(t => RootTypeCanonicalNamePrefix + t.getCanonicalName)
+
+  // todo I hate myself :<
+  private[monads] def isStatefulMonadType(t: ScType): Boolean = {
+    val psiEl = t.unpackedType.asInstanceOf[ScDesignatorType].element.asInstanceOf[PsiClass]
+    val superTypes = psiEl.getImplementsListTypes
+    val selfTypeAndSuperTypesNames = superTypes.map(_.getCanonicalText).map(t => RootTypeCanonicalNamePrefix + cleanTypeParams(t)) :+ cleanTypeParams(t.canonicalText)
+    selfTypeAndSuperTypesNames.exists(StatefulMonadsTypesNames.contains)
```

Comment:
Yep, my refactoring was not well suited for your task. Now you can pass ScType:
https://github.com/JetBrains/intellij-scala/commit/d2de9f714086f253b2754ffcd11a82f1f101947a

The problem with your method is that you are trying to check conformance on the level of psi elements instead of using methods of ScType. It's the key purpose of our model of scala type system, after all.
PS. This discussion may be of interest for other people, who cannot speak our beautiful language :)